### PR TITLE
[incubator/jaeger] Update image, add service accounts, remove configmap, passwords as secrets

### LINKED ISSUE #15122: [incubator/jaeger] passwords should be in a secret and used via secretKeyRef.
Please see title.

### PR DESCRIPTION
**Pull request by naseemkullah**
* Update image to 1.13.0
* Use service accounts for all components
* Configure storage via env vars directly, remove configmap
* Configure db passwords as secrets, via secretKeyRef
* Add option for existingSecret for db passwords

* fixes #15122
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: incubator/jaeger/templates/spark-cronjob.yaml [hunk 1]
```diff
@@ -44,37 +45,19 @@ spec:
...
-                configMapKeyRef:
-                  name: {{ include "jaeger.fullname" . }}
-                  key: es.nodes-wan-only
+              value: {{ .Values.storage.elasticsearch.nodesWanOnly | quote }}
             - name: ES_PASSWORD
-              valueFrom:
-                configMapKeyRef:
-                  name: {{ include "jaeger.fullname" . }}
-                  key: es.password
+              value: {{ .Values.storage.elasticsearch.password }}
```
Comment (monotek) [inline incubator/jaeger/templates/spark-cronjob.yaml]: Imho passwords should be in a secret and used via secretKeyRef.

File: incubator/jaeger/Chart.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
-appVersion: 1.12.0
+appVersion: 1.13.0
```
Comment (monotek) [inline incubator/jaeger/Chart.yaml]: Is removing the configmaps a breaking change for users using the current version?
If so version should be 2.0.0 and maybe the chart should be moved to stable at the same time.

**Review by monotek — CHANGES_REQUESTED** [review]

File: incubator/jaeger/Chart.yaml (hunk 2 above)
Comment (naseemkullah) [reply incubator/jaeger/Chart.yaml]: It is not a breaking change to my knowledge. In any case this is the app version. The chart version is still <1.0.0. The chart will make its way to stable soon enough but not quite in this PR.

File: incubator/jaeger/templates/spark-cronjob.yaml (hunk 1 above)
Comment (naseemkullah) [reply incubator/jaeger/templates/spark-cronjob.yaml]: Agreed. I will make an issue for this and tackle it in a separate PR. Thanks.

Comment (naseemkullah) [reply incubator/jaeger/templates/spark-cronjob.yaml]: https://github.com/helm/charts/issues/15122 created. I encourage you to submit a PR for this @monotek

Comment (naseemkullah) [reply incubator/jaeger/templates/spark-cronjob.yaml]: Agreed, #15122 created, let me try to include the fix in this PR.

Comment (naseemkullah) [reply incubator/jaeger/templates/spark-cronjob.yaml]: Changes applied.

**Comment by maorfr** [comment]
Found this section regarding service account: https://helm.sh/docs/chart_best_practices/#using-rbac-resources

**Comment by naseemkullah** [comment]
Thanks I’ll take a look and re adjust accordingly

**Comment by naseemkullah** [comment]
@maorfr I've refactored the service accounts. Please let me know if it seems good to you.

**Comment by maorfr** [comment]
/lgtm

thanks for your on-going contributions!

**Comment by naseemkullah** [comment]
My pleasure @maorfr !

**Pull request merged**